### PR TITLE
Add .editorconfig

### PR DESCRIPTION
**Pull request by samruddhikhandale**
Addresses https://github.com/devcontainers/images/pull/475#issuecomment-1456149887
From https://github.com/microsoft/vscode-dev-containers/blob/main/.editorconfig

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[*.json]
```
Comment (eitsupi) [inline .editorconfig:1]: In general, it seems that `root = true` is added to the project root file.
https://github.com/editorconfig/editorconfig/blob/b52217517b2c5b2c3b6a0daa7260983d409e7e6b/.editorconfig#LL2-L2C12

```suggestion
root = true

[*.json]
```

**Review by eitsupi — COMMENTED** [review]
It would be useful to add the editorconfig extension for VS Code to the devcontainer.json.
https://github.com/microsoft/vscode-dev-containers/blob/5a084a93b0736ea86395ac99019a5b72a00b6341/.vscode/extensions.json#L6-L9

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -1,23 +1,17 @@
...
                 "mads-hartmann.bash-ide-vscode",
-                "dbaeumer.vscode-eslint"
+                "dbaeumer.vscode-eslint",
+                "EditorConfig.EditorConfig"
             ]
-        },
-        "codespaces": {
-            "repositories": {
-                "devcontainers/features": {
-                    "permissions": "write-all"
```
Comment (samruddhikhandale) [inline .devcontainer/devcontainer.json:13]: I don't believe we need these permissions anymore. @joshspicer is that correct?

Comment (joshspicer) [reply .devcontainer/devcontainer.json:13]: Not necessary.  It may be handy at times but definitely ok to remove.

**Review by joshspicer — APPROVED** [review]

**Pull request merged**